packages/rule-tester/src/RuleTester.ts

```diff
@@ -573,29 +572,43 @@ export class RuleTester extends TestFramework {
     // Verify the code.
     // @ts-expect-error -- we don't define deprecated members on our types
     const { getComments } = SourceCode.prototype as { getComments: unknown };
-    let messages;
-
-    try {
-      // @ts-expect-error -- we don't define deprecated members on our types
-      SourceCode.prototype.getComments = getCommentsDeprecation;
-      messages = this.#linter.verify(code, config, filename);
-    } finally {
-      // @ts-expect-error -- we don't define deprecated members on our types
-      SourceCode.prototype.getComments = getComments;
-    }
 
-    const fatalErrorMessage = messages.find(m => m.fatal);
+    let initialMessages: Linter.LintMessage[] | null = null;
+    let messages: Linter.LintMessage[] | null = null;
+    let fixedResult: SourceCodeFixer.AppliedFixes | null = null;
+    let passNumber = 0;
+    const outputs: string[] = [];
 
-    assert(
-      !fatalErrorMessage,
-      `A fatal parsing error occurred: ${fatalErrorMessage?.message}`,
-    );
+    do {
+      passNumber++;
 
-    // Verify if autofix makes a syntax error or not.
-    if (messages.some(m => m.fix)) {
-      output = SourceCodeFixer.applyFixes(code, messages).output;
+      try {
+        // @ts-expect-error -- we don't define deprecated members on our types
+        SourceCode.prototype.getComments = getCommentsDeprecation;
+        messages = this.#linter.verify(code, config, filename);
+        if (!initialMessages) {
+          initialMessages = messages;
+        }
+      } finally {
+        // @ts-expect-error -- we don't define deprecated members on our types
+        SourceCode.prototype.getComments = getComments;
+      }
+
+      const fatalErrorMessage = messages.find(m => m.fatal);
+      assert(
+        !fatalErrorMessage,
+        `A fatal parsing error occurred: ${fatalErrorMessage?.message}`,
+      );
+
+      fixedResult = SourceCodeFixer.applyFixes(code, messages);
+      if (!messages.length || fixedResult.output === code) {
+        continue;
+      }
+      outputs.push((code = fixedResult.output));
```

Comment:
(Style preference, optional) 
```suggestion
      code = fixedResult.output;
      outputs.push(code);
```